src/Appwrite/Utopia/Response.php

```diff
@@ -556,7 +559,11 @@ public function dynamic(Document $document, string $model): void
 
         switch ($this->getContentType()) {
             case self::CONTENT_TYPE_JSON:
-                $this->json(!empty($output) ? $output : new \stdClass());
+                try {
+                    $this->json(!empty($output) ? $output : new \stdClass());
+                } catch (JsonException $e) {
+                    throw new Exception('Failed to parse binary response: ' . $e->getMessage(), 400);
```

Comment:
What should be the error message for this be called @christyjacob4? And is it okay for it to be a 400 error?

Reply:
The exception can arise in case of any malformed JSON . So the error message shouldn't talk about binary response specifically. 
400 is fine.